### PR TITLE
chore: Create renovate_config_check.yaml

### PR DESCRIPTION
**Pull request by ddixit14**
Introducing config checker for renovate bot upon every pull request touching `renovate.json`.

### CODE REVIEW

File: synthtool/gcp/templates/java_library/.github/workflows/renovate_config_check.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Renovate Bot Config Validation
+
+on:
+  pull_request:
+    paths:
+      - 'renovate.json'
+
+jobs:
+  config_validation:
+    runs-on: ubuntu-latest
```
Comment (suztomo) [inline synthtool/gcp/templates/java_library/.github/workflows/renovate_config_check.yaml:10]: Can you set a specific version? I don't want to see failure due to the "latest" starting to point a new version.

File: synthtool/gcp/templates/java_library/.github/workflows/renovate_config_check.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: Renovate Bot Config Validation
+
+on:
+  pull_request:
+    paths:
+      - 'renovate.json'
+
+jobs:
+  config_validation:
```
Comment (suztomo) [inline synthtool/gcp/templates/java_library/.github/workflows/renovate_config_check.yaml:9]: Include renovate bot in its name.

File: synthtool/gcp/templates/java_library/.github/workflows/renovate_config_check.yaml (hunk 1 above)
Comment (ddixit14) [reply synthtool/gcp/templates/java_library/.github/workflows/renovate_config_check.yaml]: Done.

File: synthtool/gcp/templates/java_library/.github/workflows/renovate_config_check.yaml (hunk 2 above)
Comment (ddixit14) [reply synthtool/gcp/templates/java_library/.github/workflows/renovate_config_check.yaml]: Done.

**Review by suztomo — APPROVED** [review]

**Comment by suztomo** [comment]
If you go for this approach, ensure the pull requests in the downstream repositories are merged.

**Pull request merged**